usagov/usagov-2021
# Usagov 1752 refactor datalayer

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1752

## Description


## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



You can use `scripts/drush/veryify-pagelayer-data.php` to validate that your changes don't change the datalayer output. This is more useful for refactoring the event subscribers or ensuring that additions to the data don't introduce unintended changes. Using the script takes some preparation:

1. *Create Reference Output* The script compares Drupal outputs against a "known-good" data set, the published page CSV file created by a tome export. To prepare this file, you should export your local site using the code from the `dev` branch.
2. *Add to the datalayer* In a new branch, create an event listener to make your changes.
3. *Run verify script* use `drush php:script scripts/drush/verify-pagelayer-data.php`. It will fetch pages from your local site and compare the output of your datalayer to the expected values in the CSV file. If it detects any difference in values, it'll echo the actual versus expected values.

The verify script takes two optional options, specified after `--`. Use `--base` if your local hostname is not reachable via `http://localhost/`. Ise the `--sample` flag for a quicker run that doesn't check every URL in the CSV file.

```sh
drush php:script scripts/drush/verify-pagelayer-data.php \
  -- --base https://example.com/ \
  --sample
```

### Change Requirements



### Validation Steps

1. See testing instructions for running the verification script. Run it and pipe outout to a text file. Confirm the script doesn't detect any changes in the datalayer values.


## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
